README.md

```diff
@@ -1,182 +1,163 @@
-
-JavaScript can't do that. It sees `2345` and only knows that it's a number. Similarly, it sees _anything_ in quotation marks as a string. We provide the context for JavaScript according to the rules that JavaScript follows — one of those rules happens to describe what happens when we add a number and a string.
-
-## You're just not my type
-
-How do we know what types we're dealing with? JavaScript gives us the handy dandy `typeof`. We use it like so
-
-``` javascript
-typeof 2 // "number"
-typeof "2" // "string"
-typeof '2' // "string" — strings can be enclosed in single (') or double (") quotes
+Even empty strings are strings:
+```js
+typeof ""
+//=> "string"
 ```
 
-Pretty simple, right? You can think of `typeof` as a shortcut for telling you what kind of hole to look for in our imaginary, new-fangled shape-sorter.
-
-In addition to numbers and strings, JavaScript has the following _primitive_ types:
-
-- Boolean
-- Undefined
-- Null
-
-Enter the following commands in your console to get a feel for the different types:
+### Booleans
+As in Ruby, booleans can only be the logical values `true` or `false`. Booleans play a big role in control flows and looping in JavaScript.
+```js
+typeof true
+//=> "boolean"
 
-``` javascript
-typeof 1
-typeof 10
-typeof 1.123
+typeof false
+//=> "boolean"
 ```
 
-``` javascript
-typeof "Albert"
-typeof '123'
-typeof "What's my type?"
-```
+### Symbols
+Symbols are a relatively new data type (introduced in ES2015) that's primarily used as a way to add anonymous properties to objects. Don't worry about symbols for now.
 
-``` javascript
-typeof true
-typeof false
-```
+### Objects
+JavaScript objects are a collection of properties bounded by curly braces (`{ }`), similar to a hash in Ruby. The properties can point to values of any data type — even other objects:
+```js
+{
+  "name": "JavaScript",
+  "createdBy": {
+    "firstName": "Brendan",
+    "lastName": "Eich"
+  },
+  "firstReleased": 1995,
+  "isAwesome": true
+}
 
-``` javascript
-typeof undefined
+typeof {}
+//=> "object"
 ```
 
-``` javascript
+### `null`
+The `null` data type represents an intentionally absent object. For example, if a piece of code returns an object when it successfully executes, we could have it return `null` in the event of an error. Confusingly, the `typeof` operator returns `"object"` when called with `null`:
+```js
 typeof null
+//=> "object"
 ```
 
-Also enter the following — pay attention to the errors!
-
-``` javascript
-typeof 1.123.45
+### `undefined`
+The bane of many JS developers, `undefined` is a bit of a misnomer. Instead of 'not defined,' it actually means something more like 'not yet assigned a value.' We'll learn a lot more about `undefined` when we dive into variables and functions.
+```js
+typeof undefined
```

Comment:
@jmburges: https://github.com/learn-co-curriculum/javascript-intro-to-variable-assignment-and-declaration/blob/wip-master/README.md#variable-values